dashboard/app/models/script.rb

```diff
@@ -136,11 +136,29 @@ def self.script_cache_from_db
   end
 
   def self.script_cache
-    return nil if Rails.application.config.levelbuilder_mode # cache disabled when building levels
+    return nil if Rails.application.config.levelbuilder_mode
     @@script_cache ||=
       script_cache_from_cache || script_cache_from_db
   end
 
+  # Returns a cached map from script level id to id.
+  def self.script_level_cache
+    @@script_level_cache ||= {}.tap do |cache|
+      script_cache.values.each do |script|
+        cache.merge!(script.script_levels.index_by(&:id))
+      end
+    end
+  end
+
+  def self.cache_find_script_level(level_id)
```

Comment:
maybe rename this arg to script_level_id to avoid confusion with the level_id field in ScriptLevel